samples/settings/src/settings.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2021 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <sys/printk.h>
+#include <init.h>
+#include <ctype.h>
+
+#include <settings/settings.h>
+
+bool json_output;
```

Comment:
Defining this variable at the function scope i cannot access it from another function.

In my specific case, i can define a local scope for the bool_json at the `cmd_settings_set` function.

But for the other 2 functions, `cmd_settings_get` and `settings_read_callback`, i did not find a way to do this because i need to access the boolean in the `settings_read_callback` and this function is called by the first, `cmd_settings_get` as a callback in a third function and, therefore, i cannot pass the boolean to the second function because i don't have control over the callback.

If i'm wrong could you help with any suggestions about it? Maybe i'm not seeing anything easy that you can see.

Reply:
One other potential problem is that `params ` may no longer be in scope when `settings_read_callback` gets called. So I'd recommend making the struct instance static:

```c
	static struct settings_read_callback_params params = {
		.shell_ptr = shell,
		.value_found = false,
                .json_output = json_output
	};
```